src/providers/ad/ad_common.c

```diff
@@ -1305,6 +1305,21 @@ ad_get_auth_options(TALLOC_CTX *mem_ctx,
     DEBUG(SSSDBG_CONF_SETTINGS, "Option %s set to %s\n",
           krb5_options[KRB5_USE_KDCINFO].opt_name,
           ad_opts->service->krb5_service->write_kdcinfo ? "true" : "false");
+    if (ad_opts->service->krb5_service->write_kdcinfo) {
+        ret = sss_krb5_parse_lookahead(
```

Comment:
sss_krb5_parse_lookahead is a void function, so it makes little sense to check its return value, it's actually confusing because the caller might think there are some checks done.

Additionally, would it make sense to split this block into a generic function somewhere in krb5_common.c or elsewhere that would accept krb5_service and krb5_options and set the members of krb5_service on its own? Maybe the parse function could then really just parse the option and not set a fallback and the fallback (-1) could be set in this utility function.

Reply:
Sorry, I was reviewing the patches one by one, this is actually partly solved in the last code patch by removing the errno check